Source/WebCore/html/HTMLMediaElement.cpp

```diff
@@ -5259,6 +5265,11 @@ void HTMLMediaElement::sourceWasAdded(HTMLSourceElement& source)
         return;
     }
 
+#if ENABLE(MEDIA_SOURCE)
+    if (m_mediaSource)
+        m_mediaSource->openIfDeferredOpen();
```

Comment:
Do we want to change the state if _any_ `<source>` is added, or only if it is one that is AirPlay-compatible?

Reply:
the end result is the same, and it allows to have management to when to allow MMS in a single place only

Reply:
`openIfDeferredOpen` unconditionally sets the `readyState` to open, so won't the following enable MMS even though the new source element is useless?

    video.appendChild(document.createElement('source'))

Reply:
`openIfDeferredOpen` doesn't sets the readyState to open. It first checks if ManagedMediaSource::isOpen() returns true; which it won't unless the right conditions are matched.

Reply:
> video.appendChild(document.createElement('source'))

The code checks that there's a source, and that the src of that source is usable.